plasmapy/diagnostics/proton_radiography.py

```diff
@@ -300,6 +305,212 @@ def _log(self, msg):
     # Define some constants so they don't get constantly re-evaluated
     _c = const.c.si.value
 
+    # *************************************************************************
+    # Create mesh
+    # *************************************************************************
+
+    def add_wire_mesh(
+        self, location, extent, nwires, wire_diameter, mesh_hdir=None, mesh_vdir=None
+    ):
+        """
+        Add a wire mesh grid between the particle source and the object grid
+        that blocks particles which intersect the wires.
+
+        Parameters
+        ----------
+        location : `~astropy.units.Quantity`, shape (3)
+            A vector pointing from the origin of the grid to the center of the
+            mesh grid. This location must be between the source and the
+            object grid.
+
+            This vector will be interpreted as
+            being in either cartesian, cylindrical, or spherical coordinates
+            based on its units. Valid geometries are:
+
+            * Cartesian (x,y,z) : (meters, meters, meters)
+            * cylindrical (r, theta, z) : (meters, radians, meters)
+            * spherical (r, theta, phi) : (meters, radians, radians)
+
+            In spherical coordinates theta is the polar angle.
+
+        extent : Tuple of 1 or 2 `~astropy.units.Quantity`
+            The size of the mesh grid (in the mesh plane). If one value
+            is provided, the mesh is circular and the value provided is
+            interpreted as the diameter. If two values are provided, the
+            mesh is rectangular and they the values are interpreted as the
+            width and height respectively.
+
+        nwires : Tuple of 1 or 2 ints
```

Comment:
Could also be a single int for convenience, but this is fine. Oh, you cast it to tuple later - great! Just note it down here.